### PR TITLE
feat: initial apollo configutation

### PR DESCRIPTION
**Pull request by cbachmeier**
Initial Apollo setup

https://uniswaplabs.atlassian.net/browse/WEB-2582

### CODE REVIEW

File: codegen.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { CodegenConfig } from '@graphql-codegen/cli'
+
+const config: CodegenConfig = {
+  overwrite: true,
+  schema: './src/graphql/data/schema.graphql',
+  documents: ['./src', '!**/node_modules/**', '!**/__mocks__/**', '!**/__generated__/**', '!**/thegraph/**'],
+  generates: {
+    'src/graphql/data/__generated__/types-and-hooks.ts': {
```
Comment (justindomingue) [inline codegen.ts:8]: they recommend checking in this generated file

Comment (cbachmeier) [reply codegen.ts:8]: Currently we don't check in any of the generated files, what are your thoughts @zzmp ?

Comment (cbachmeier) [reply codegen.ts:8]: In latest it is checked in

Comment (cbachmeier) [reply codegen.ts:8]: One thing to note is that the checked in schema file would need to be ignored by our linting

Comment (zzmp) [reply codegen.ts:8]: in favor of checking it in.

File: codegen.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import type { CodegenConfig } from '@graphql-codegen/cli'
```
Comment (zzmp) [inline codegen.ts:1]: Can this file be renamed to `apollo-codegen.ts`?

File: codegen.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+const config: CodegenConfig = {
+  overwrite: true,
+  schema: './src/graphql/data/schema.graphql',
+  documents: ['./src', '!**/node_modules/**', '!**/__mocks__/**', '!**/__generated__/**', '!**/thegraph/**'],
```
Comment (zzmp) [inline codegen.ts:6]: can this be more specific?

File: codegen.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import type { CodegenConfig } from '@graphql-codegen/cli'
+
+const config: CodegenConfig = {
```
Comment (zzmp) [inline codegen.ts:3]: Please link to the docs explaining what this file does.

File: codegen.ts (hunk 3 above)
Comment (cbachmeier) [reply codegen.ts:6]: sure, but this was a direct port of what was in `relay.config.js`

**Pull request merged**